ui/src/overview.tsx

```diff
@@ -129,11 +139,89 @@ function ParticipantsGroup(props: { group: Group }) {
   );
 }
 
+type RenameCriteriaDialogProps = {
+  callback: (name: string) => void;
+  name: string;
+};
+
+function useRenameCriteriaDialog(
+  props: RenameCriteriaDialogProps
+): [ReactNode, () => void] {
+  const [open, setOpen] = useState(false);
+  const show = () => {
+    setOpen(true);
+  };
+
+  const [name, setName] = useState(props.name);
+  const onNameChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setName(event.target.value);
+  };
+
+  const onCreate = () => {
+    setOpen(false);
+    props.callback(name);
+  };
+
+  return [
+    // eslint-disable-next-line react/jsx-key
```

Comment:
This is copy from the <Dialog> from dataset.tsx file

Reply:
Hi Tim. While I am doing it. I have a question. Should I modify the dataset.tsx file's <dialog> also use the new created dialog module in the same branch? Because I remember, last time you said we should make sure the commit be clean and only contain the stuffs related to the same feature. But I think if I do not do this here, we will need create another branch/pr individually for the dataset.tsx file. It might be too light changes?

Reply:
That is a good thought. There are no hard rules here though. It's a question of balancing the overhead of multiple PRs with the ability to examine/revert/merge previous changes and the difficulty of reviewing them. If this change was really large I would suggest separating the `dataset.tsx` changes to make it easier to review. On the other hand, fixing several unrelated typos in the same PR wouldn't be unreasonable. Sharing the code would be a significant part of this change so I would leave them together. You could certainly make an argument that it's better to change the existing `dataset.tsx` usage in a separate PR though. Having small PRs isn't a problem as long as they're a complete chunk of work and it's better than combining unrelated changes.

Reply:
I correct it in this commit while correct the comments